app/assets/javascripts/discourse/app/models/group.js

```diff
@@ -127,6 +127,14 @@ const Group = RestModel.extend({
     });
   },
 
+  join() {
+    return ajax(`/groups/${this.id}/join.json`, {
+      type: "PUT",
+    }).then(() => {
+      this.findMembers();
```

Comment:
Are you purposely swallowing the promise here? `join()` will return before `findMembers` is finished. I'd recommend adding `return` or using the one line version: `}).then(() => this.findMembers());`

Reply:
Thoughtlessly copied this from neighbor methods. I've fixed it only in this method by now, but I'll take a look and fix other methods a bit later too.